job/server/src/main/java/alluxio/job/plan/PlanDefinition.java

```diff
@@ -44,10 +46,10 @@
    * @param config the job configuration
    * @param jobWorkerInfoList the list of available workers' information
    * @param selectExecutorsContext the context containing information used to select executors
-   * @return a map of selected workers to the parameters to pass along
+   * @return a set of pairs of selected workers to the parameters to pass along
    * @throws Exception if any error occurs
    */
-  Map<WorkerInfo, P> selectExecutors(T config, List<WorkerInfo> jobWorkerInfoList,
+  Set<Pair<WorkerInfo, P>> selectExecutors(T config, List<WorkerInfo> jobWorkerInfoList,
```

Comment:
Is this really the best way to represent the output? It seems more natural to me to map the `WorkerInfo` to another collection, rather than by using `Pair`s

I'm not opposed - just wondering if you thought of other ways of representing it and why you landed on this

Reply:
Yes. The primary reason is because P can also be a collection.